flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/source/log/pulsar/LogMsgWithRetractInfo.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.flink.read.source.log.pulsar;
+
+import com.netease.arctic.flink.read.source.log.kafka.LogKafkaPartitionSplit;
+import com.netease.arctic.log.LogData;
+import org.apache.flink.connector.pulsar.source.reader.message.PulsarMessage;
+import org.apache.pulsar.client.api.MessageId;
+
+public class LogMsgWithRetractInfo<T> extends PulsarMessage<T> {
```

Comment:
```suggestion
public class LogRecordPulsarWithRetractInfo<T> extends PulsarMessage<T> {
```
Besides, LogRecordWithRetractInfo renames to LogRecordKafkaWithRetractInfo, keeping the same pattern with the Pulsar retract record class name.